lib/wibox/layout/ratio.lua

```diff
@@ -96,18 +101,75 @@ function ratio:layout(_, width, height)
             w, h = floor(space), height
         end
 
-        table.insert(result, base.place_widget_at(v, x, y, w, h))
+        -- Keep track of the unused entries
+        if has_stragety then
+            fit_h, fit_w = base.fit_widget(
+                self, context, v,
+                dir == "x" and floor(space) or w,
+                dir == "y" and floor(space) or h
+            )
+
+            is_void = (v.visible == false)
```

Comment:
If `visible == false`, `wibox.widget.base.fit_widget` will return `0, 0`. Thus, is this first part of the `or` really needed? I think the rest of the check should be enough.